modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/dbf/index/DbaseIndexManager.java

```diff
@@ -0,0 +1,199 @@
+        
+        // Check if all properties in filter are indexed.
+        for (PropertyName property : visitor.propertiesMap.values()) {
+            if (!indexNodeMap.containsKey(property.getPropertyName()) && !geometryName.equalsIgnoreCase(property.getPropertyName())) {
+                return false;
+            }
+        }
+        return true;
+    }
+    
+    /**
+     * Converts the specified collection of Dbase record numbers to a list of Data Objects.
+     * @throws IOException
+     */
+    private List<Data> convertRecordNumbersToDataObjects(Collection<Integer> recnoCollection, boolean clearCollection) throws IOException {
+        
+        if (recnoCollection != null) {
+            IndexFile shx = shpFiles.exists(SHX) && shpFiles.get(SHX) != null ? new IndexFile(shpFiles, store.isMemoryMapped()) : null;
+            
+            DataDefinition def = new DataDefinition("US-ASCII");
+            def.addField(Integer.class);
+            def.addField(Long.class);
+            
+            try {
+                List<Data> records = new ArrayList<Data>(recnoCollection.size());
+                
+                for (int recno : recnoCollection) {
+                    
+                    Data data = new Data(def);
+                    data.addValue(new Integer(recno + 1));
+                    data.addValue(new Long(shx.getOffsetInBytes(recno)));
+                    
+                    records.add(data);
+                }
+                if (clearCollection) {
+                    recnoCollection.clear();
+                }
+                
+                // Sort offsets, it is faster to read from disk.
+                if (records.size() > 1) {
+                    Collections.sort(records, new Comparator<Data>() {
+                        public int compare( Data data1, Data data2 ) {
+                            return ((Long)data1.getValue(1)).compareTo((Long)data2.getValue(1));
+                        }
+                    });
+                }                
+                return records;
+                
+            } finally {
+                shx.close();
+            }
+        }
+        return null;
+    }
+    
+    /**
+     * Queries a supported Dbase index for features available in the specified alphanumeric filter.
+     * 
+     * @param filter Alphanumeric Filter to use
+     * @param maxFeatures Maximum number of features to fetch
+     * @return a list of Data objects 
+     * @throws IOException
+     */
+    public List<Data> queryDbaseIndex(Filter filter, int maxFeatures) throws IOException {
+        
+        if (filter!=null && !Filter.INCLUDE.equals(filter) && !Filter.EXCLUDE.equals(filter)) {
+            
+            DbaseFileIndex dbaseIndex = null;
+            
+            // Exist a supported index manager for the current Dbase file?
+            if (!(filter instanceof BinarySpatialOperator)) {
```

Comment:
This code verifies if the filter is a BinarySpatialOperator because in this case it is not necessary search whether a dbase-index file exist. It is a pure spatial filter, and then, the code only will use the bbox defined.